### PR TITLE
Add gas overrides before sending a transaction

### PR DESCRIPTION
**Pull request by iturricf**
In particular we are interested in adding polygon overrides

### CODE REVIEW

**Review by scnale — APPROVED** [review]

**Comment by abhidtu2014** [comment]
Hey @Cristian Iturri 
I am reviewing this PR, don't merge now, there are some nit/improvements I would like to suggest

File: src/relayer/cctp.relayer.ts [hunk 1]
```diff
@@ -326,3 +332,18 @@ export class CctpRelayer {
...
+
+  const overrides: PayableOverrides = {};
+
+  if (ov.gasPrice) Object.assign(overrides, { gasPrice: ov.gasPrice });
+  if (ov.type) Object.assign(overrides, { type: ov.type });
+  if (ov.maxFeePerGas)
+    Object.assign(overrides, { maxFeePerGas: ov.maxFeePerGas });
+  if (ov.maxPriorityFeePerGas)
+    Object.assign(overrides, { maxPriorityFeePerGas: ov.maxPriorityFeePerGas });
+
```
Comment (abhidtu2014) [inline src/relayer/cctp.relayer.ts:347]:
```suggestion
  const {gasPrice, type, maxFeePerGas, maxPriorityFeePerGas } = await ctx.calculateOverridesForChain(toChainId);

  const overrides: PayableOverrides = {
  ...(gasPrice & { gasPrice }),
  ...(type && { type }),
  ...(maxFeePerGas && { maxFeePerGas }),
  ...(maxPriorityFeePerGas && { maxPriorityFeePerGas })
  };
```

Comment (abhidtu2014) [reply src/relayer/cctp.relayer.ts:347]: We can make look little better.

**Review by abhidtu2014 — COMMENTED** [review]
LGTM 👍 

Just added a nit comment.

File: src/relayer/cctp.relayer.ts [hunk 2]
```diff
@@ -326,3 +332,15 @@ export class CctpRelayer {
...
 }
+
+async function buildOverrides(ctx: CctpRelayerContext, toChainId: EVMChainId) {
+  const {gasPrice, type, maxFeePerGas, maxPriorityFeePerGas } = await ctx.calculateOverridesForChain(toChainId);
+
+  const overrides = {
+  ...(gasPrice && { gasPrice }),
+  ...(type && { type }),
+  ...(maxFeePerGas && { maxFeePerGas }),
+  ...(maxPriorityFeePerGas && { maxPriorityFeePerGas })
```
Comment (scnale) [inline src/relayer/cctp.relayer.ts:343]:
```suggestion
    ...(gasPrice && { gasPrice }),
    ...(type && { type }),
    ...(maxFeePerGas && { maxFeePerGas }),
    ...(maxPriorityFeePerGas && { maxPriorityFeePerGas })
```

**Review by solanoepalacio — APPROVED** [review]

**Pull request merged**